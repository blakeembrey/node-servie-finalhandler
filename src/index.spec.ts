import { Request } from "servie/dist/node";
import { finalhandler } from "./index";

describe("servie-finalhandler", () => {
  it("should return a 404 response", () => {
    const req = new Request("/");

    return finalhandler(req)().then(res => expect(res).toMatchSnapshot());
  });

  it("should not return a body when using HEAD", () => {
    const req = new Request("/", { method: "head" });

    return finalhandler(req)().then(res => expect(res).toMatchSnapshot());
  });
});
