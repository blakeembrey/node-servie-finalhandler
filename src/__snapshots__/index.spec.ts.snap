// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`servie-finalhandler should not return a body when using HEAD 1`] = `
Object {
  "body": Object {
    "bodyUsed": false,
    "buffered": true,
    "hasBody": false,
    "headers": Object {},
  },
  "bytesTransferred": 0,
  "finished": false,
  "headers": Object {
    "content-security-policy": "default-src 'self'",
    "x-content-type-options": "nosniff",
  },
  "started": false,
  "statusCode": 404,
  "statusMessage": undefined,
  "trailers": Object {},
}
`;

exports[`servie-finalhandler should return a 404 response 1`] = `
Object {
  "body": Object {
    "bodyUsed": false,
    "buffered": true,
    "hasBody": true,
    "headers": Object {
      "content-length": "12",
      "content-type": "text/plain",
    },
  },
  "bytesTransferred": 0,
  "finished": false,
  "headers": Object {
    "content-security-policy": "default-src 'self'",
    "x-content-type-options": "nosniff",
  },
  "started": false,
  "statusCode": 404,
  "statusMessage": undefined,
  "trailers": Object {},
}
`;
